File: MS.NET/Fundamentals/Language/OOP/interfacetest1.cs
using Banking;
using System;
using static Banking.Banker;

static class Program
{
	private static void DeductTax(Account[] accounts)
	{
		foreach(Account entry in accounts)
		{
			ITaxable t = entry as ITaxable;
			t?.Deduct(); //if(t != null) t.Deduct()
		}
	}

	private static void PayAnnualInterest(Account[] accounts)
	{
		foreach(Account entry in accounts)
		{
			IProfitable p = entry as IProfitable;
			if(p != null)
			{
				double interest = p.GetInterest(1);
				entry.Deposit(interest);
			}
		}
	}

	public static void Main()
	{
		Account[] bank = new Account[4];
		bank[0] = OpenCurrentAccount();
		bank[0].Deposit(20000);
		bank[1] = OpenSavingsAccount();
		bank[1].Deposit(25000);
		bank[2] = OpenSavingsAccount();
		bank[2].Deposit(35000);
		bank[3] = OpenCurrentAccount();
		bank[3].Deposit(50000);

		DeductTax(bank);
		PayAnnualInterest(bank);

		foreach(Account acc in bank)
			Console.WriteLine("{0}\t{1}", acc.Id, acc.Balance);

	}
}

File: Java/Fundamentals/Language/OOP/InterfaceTest1.java
import edu.met.banking.*;

class InterfaceTest1{

	private static void payAnnualInterest(Account[] accounts){
		for(Account acc : accounts){
			if(acc instanceof Profitable){
				Profitable p = (Profitable)acc;
				double amt = p.getInterest(1);
				acc.deposit(amt);
			}
		}
	}

	public static void main(String[] args){
		Account[] bank = new Account[4];
		bank[0] = Banker.openCurrentAccount();
		bank[0].deposit(10000);
		bank[1] = Banker.openSavingsAccount();
		bank[1].deposit(15000);
		bank[2] = Banker.openSavingsAccount();
		bank[2].deposit(25000);
		bank[3] = Banker.openCurrentAccount();
		bank[3].deposit(40000);
		payAnnualInterest(bank);
		for(Account acc : bank)
			System.out.printf("%d\t%.2f%n", acc.getId(), acc.getBalance());
	}
}


